router/core/graphql_prehandler.go

```diff
@@ -597,6 +602,36 @@ func (h *PreHandler) handleOperation(req *http.Request, buf *bytes.Buffer, httpO
 
 	httpOperation.traceTimings.EndPlanning()
 
+	/**
+	* Validate that the planned query doesn't exceed the maximum query depth configured
+	 */
+
+	if h.maxQueryDepth > 0 {
+		_, maxQueryDepthSpan := h.tracer.Start(req.Context(), "Operation - Checking Query Depth",
+			trace.WithSpanKind(trace.SpanKindInternal),
+			trace.WithAttributes(otel.WgEngineRequestTracingEnabled.Bool(httpOperation.traceOptions.Enable)),
+			trace.WithAttributes(attributes...),
+		)
+
+		operation := opContext.preparedPlan.operationDocument
+		definition := opContext.preparedPlan.schemaDocument
+		report := operationreport.Report{}
+		globalComplexityResult, _ := operation_complexity.CalculateOperationComplexity(operation, definition, &report)
```

Comment:
We do parse and plan the operation only once. We should do this for complexity as well.